components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -181,11 +193,17 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
                 carbonUM.addUser(user.getUserName(), user.getPassword(), null, claimsMap, null);
                 log.info("User: " + user.getUserName() + " is created through SCIM.");
 
-            } catch (UserStoreException e) {
-                throw new CharonException("Error in adding the user: " + user.getUserName() + " to the user store", e);
             }
-            return user;
+        } catch (UserStoreException e) {
+            String errMsg = e.getMessage()+ " ";
+            errMsg += "Error in adding the user: " + user.getUserName() +
+                    " to the user store..";
+            throw new CharonException(errMsg,e);
```

Comment:
Fixed in https://github.com/wso2/carbon-identity/pull/685/files#diff-a65c46bdc71b1619add63b38503075c2R196